packages/pin-input/src/use-pin-input.ts

```diff
@@ -115,22 +123,24 @@ export function usePinInput(props: UsePinInputProps = {}) {
   const domContext = useDescendants<HTMLInputElement, {}>()
   const { descendants } = domContext
 
-  const [moveFocus, setMoveFocus] = useState(true)
+  const [moveFocus, setMoveFocus] = React.useState(true)
 
   const [values, setValues] = useControllableState<string[]>({
     defaultValue: toArray(defaultValue) || [],
     value: toArray(value),
     onChange: (values) => onChange?.(values.join("")),
   })
 
-  useEffect(() => {
+  React.useEffect(() => {
     if (autoFocus) {
       const firstInput = descendants[0]
       firstInput?.element?.focus()
     }
-  }, [descendants, autoFocus])
+    // We don't want to listen for updates to `autoFocus` since it only runs initially
+    // eslint-disable-next-line
```

Comment:
It'd probably be safer to refactor this a little than to disable `exhaustive-deps`. If this were to be changed in the future it might mean we miss adding a dep, or removing a dep we no longer rely on. Not the biggest deal though, just a thought for future-proofing.

Reply:
Not sure about this one but it's something we want but `eslint` gets in the way.